### PR TITLE
ElvaRivle/Misc/SidebarData

### PR DESCRIPTION
**Pull request by ElvaRivle**
ElvaRivle/Misc/SidebarData

--------------------------------------------------------------------
 MAIN
 * name and surname saved to redux store on login
 * sidebar shows that name and surname instead of hardcoded data
 * avatar displayed on sidebar instead of hardcoded image
 * Sidebar options now have same icons as Figma
--------------------------------------------------------------------
 OTHER
 * fixed auth redux to accept and save name and surname
 * fixed some older css issues (role font wrong, active link not weight 700)
--------------------------------------------------------------------
 TODO
 * ~~show avatar instead of hardcoded image~~
 * Avatar needs to be in rectangular shape, not circular
 * ~~fix chevron~~ (?)

### CODE REVIEW

**Comment by ElvaRivle** [comment]
image

File: src/components/Avatar/Avatar.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    const renderAvatar = (name, color) => {
+        const initials = name
+            .split(' ')
+            .map((word) => word[0])
+            .join('');
+        return (
+            <Avatar
+                className=""
+                style={{
```
Comment (becarselman) [inline src/components/Avatar/Avatar.js:33]: please don't use inline style

File: src/components/Avatar/Avatar.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        let hash = 0;
+        for (let i = 0; i < name.length; i++) {
+            hash = name.charCodeAt(i) + ((hash << 5) - hash);
+        }
+        const color = (hash & 0x00ffffff).toString(16).toUpperCase();
+        const darkColor = '#' + ('00000'.substring(0, 6 - color.length) + color);
+        return darkenColor(darkColor, 20);
+    };
+
+    const darkenColor = (color, percent) => {
```
Comment (becarselman) [inline src/components/Avatar/Avatar.js:15]: you don't need to do this For avatar we can stick to one color its not problem

Comment (becarselman) [reply src/components/Avatar/Avatar.js:15]: But if you really want you can leave this, but don't use inline style

File: src/components/sidebar/View.js [hunk 3]
```diff
@@ -35,11 +36,12 @@ const Sidebar = ({actions}) => {
 
             <div className="user-profile-overview">
                 <div className="profile-picture-section">
-                    <img src={SampleProfilePicture} alt="Profile picture"/>
+                    {/*<img src={SampleProfilePicture} alt="Profile picture"/>*/}
```
Comment (becarselman) [inline src/components/sidebar/View.js]: remove commented code

File: src/components/sidebar/Container.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+const mapStateToProps = (state) => {
+    return {
+        name: state.auth.userName,
+        surname: state.auth.userSurname
+    };
+};
+
+const mapDispatchToProps = (dispatch) => ({
```
Comment (becarselman) [inline src/components/sidebar/Container.js]: if you don't have any actions you can delete mapDispatchToProps

File: src/sagas/authSaga.js [hunk 5]
```diff
@@ -6,16 +6,22 @@ import {
...
 const { open } = notification;
 
 function* loginSaga(action) {
   try {
     const response = yield call(authService.login, action.payload);
-    const { token, userId } = response;
+    const { token, userId, name, surname } = response;
     localStorage.setItem("accessToken", token);
     yield put(loginSuccess(userId));
+    yield put(setUser( {
```
Comment (becarselman) [inline src/sagas/authSaga.js:20]: this will work after login But did you try to refresh page I don't see any call that will get this data back

**Review by becarselman — CHANGES_REQUESTED** [review]

**Comment by AntMarkoLovric** [comment]
Please fix UI as it's in figma, couple of notes: 

- Avatar needs to be in rectangular shape, not circular 
- Name and surname need to be in one line, not multiple 
- Border needs to be thicker
- avatar needs to be centred 
- fix chevron

File: src/sagas/authSaga.js (hunk 5 above)
Comment (ElvaRivle) [reply src/sagas/authSaga.js:20]: will test once again, but 99% sure that persistStore saves data between refreshes

**Comment by ElvaRivle** [comment]
> Avatar needs to be in rectangular shape, not circular

Will fix this when Tarik is completely done with Avatars, then I will create custom rectangular one

> Border needs to be thicker

In figma, border is 1px, same as in code. But I set it to 2px, hopefully it's good enough now

> fix chevron

You mean small icons next to links? I used icons from ant design for those, but if it is necessary, I will use those directly from figma

**Review by AntMarkoLovric — APPROVED** [review]

**Review by becarselman — APPROVED** [review]

**Pull request merged**